team-telnyx/telnyx-node
# V2.0.0 Add Typescript support

Addresses #32

This PR will remain open while v2.0.0 is in alpha

I have labeled some files as `important` for check when reviewing. Please use the file filter to omit the "excluded files" when reviewing.


- [x] resources
- [x] base files

## Manual testing

- Check README instructions on testing and typescript
- Run `npm run lint`
- Run `npm run build`
    - Check output in `dist` folder
- Check types in `types` folder
- Run `npm run test`
    - See test report
    - See resources tested and coverage. Compare against our [Telnyx openapi spec](https://raw.githubusercontent.com/team-telnyx/openapi/refs/heads/master/openapi/spec3.json)
- to run a SDK usage examples:
    - in the root dir, run `npm link`
    - go to `examples/call-control` or `examples/messaging`
    - run `npm link telnyx`
    - run `export `TELNYX_API_KEY=<your-api-key-here>`  (this should use PROD API Key)
    - run `npm run start`
